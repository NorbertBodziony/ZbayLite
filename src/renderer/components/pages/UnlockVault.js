import React from 'react'
import Countdown from 'react-countdown-now'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router'

import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'

import PageWrapper from '../ui/PageWrapper'
import VaultUnlocker from '../../containers/VaultUnlocker'

const countdownRenderer = ({ completed }) => completed && <Redirect to='/home' />

const styles = theme => ({
  gridRoot: {
    'min-height': '100vh'
  }
})

export const UnlockVault = ({
  classes,
  locked
}) => {
  return (
    <PageWrapper>
      <Grid container justify='center' alignItems='center' className={classes.gridRoot}>
        <VaultUnlocker />
      </Grid>
      { !locked && <Countdown date={Date.now() + 1000} renderer={countdownRenderer} /> }
    </PageWrapper>
  )
}

UnlockVault.propTypes = {
  classes: PropTypes.object.isRequired,
  locked: PropTypes.bool.isRequired
}

UnlockVault.defaultProps = {
  locked: true
}

export default withStyles(styles)(UnlockVault)
