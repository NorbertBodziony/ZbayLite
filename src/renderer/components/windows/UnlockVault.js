import React from 'react'
import Countdown from 'react-countdown-now'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router'

import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'

import WindowWrapper from '../ui/WindowWrapper'
import VaultUnlocker from '../../containers/VaultUnlocker'

const countdownRenderer = ({ completed }) => completed && <Redirect to='/main' />

const styles = theme => ({
  gridRoot: {
    'min-height': '100vh'
  }
})

export const UnlockVault = ({
  classes,
  locked
}) => {
  return (
    <WindowWrapper>
      <Grid container justify='center' alignItems='center' className={classes.gridRoot}>
        <VaultUnlocker />
      </Grid>
      { !locked && <Countdown date={Date.now() + 1000} renderer={countdownRenderer} /> }
    </WindowWrapper>
  )
}

UnlockVault.propTypes = {
  classes: PropTypes.object.isRequired,
  locked: PropTypes.bool.isRequired
}

UnlockVault.defaultProps = {
  locked: true
}

export default withStyles(styles)(UnlockVault)
