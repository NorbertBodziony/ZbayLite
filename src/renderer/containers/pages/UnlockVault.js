import { connect } from 'react-redux'

import UnlockVault from '../../components/pages/UnlockVault'
import vaultSelectors from '../../store/selectors/vault'

export const mapStateToProps = state => ({
  locked: vaultSelectors.locked(state)
})

export default connect(mapStateToProps)(UnlockVault)
