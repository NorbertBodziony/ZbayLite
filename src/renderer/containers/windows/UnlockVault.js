import { connect } from 'react-redux'

import UnlockVault from '../../components/windows/UnlockVault'
import vaultSelectors from '../../store/selectors/vault'

export const mapStateToProps = state => ({
  locked: vaultSelectors.locked(state)
})

export default connect(mapStateToProps)(UnlockVault)
