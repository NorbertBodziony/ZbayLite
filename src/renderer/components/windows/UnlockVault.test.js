/* eslint import/first: 0 */
jest.mock('../../containers/widgets/VaultUnlockerForm', () => {
  const VaultUnlocker = () => (<div>VaultUnlockerForm</div>)
  return VaultUnlocker
})
import React from 'react'
import { shallow } from 'enzyme'

import { UnlockVault } from './UnlockVault'
import { mockClasses } from '../../../shared/testing/mocks'

describe('UnlockVault', () => {
  const originalDate = Date.now

  beforeAll(() => {
    global.Date.now = jest.fn(() => 1536662638084)
  })

  afterAll(() => {
    global.Date.now = originalDate
  })

  it('renders component', () => {
    const result = shallow(
      <UnlockVault
        classes={mockClasses}
      />
    )
    expect(result).toMatchSnapshot()
  })

  it('renders unlocked', () => {
    const result = shallow(
      <UnlockVault
        classes={mockClasses}
        locked={false}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
