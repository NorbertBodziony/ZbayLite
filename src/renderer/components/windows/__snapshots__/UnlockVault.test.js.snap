// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnlockVault renders component 1`] = `
<Memo(WithStyles(WindowWrapper))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="gridRoot-mock"
    container={true}
    justify="center"
  >
    <VaultUnlocker />
  </WithStyles(ForwardRef(Grid))>
</Memo(WithStyles(WindowWrapper))>
`;

exports[`UnlockVault renders unlocked 1`] = `
<Memo(WithStyles(WindowWrapper))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="gridRoot-mock"
    container={true}
    justify="center"
  >
    <VaultUnlocker />
  </WithStyles(ForwardRef(Grid))>
  <Countdown
    date={1536662639084}
    renderer={[Function]}
  />
</Memo(WithStyles(WindowWrapper))>
`;
